File: Backend/src/main/java/com/project/consphere/service/CommentService.java
package com.project.consphere.service;

import com.project.consphere.dto.CommentResponse;
import com.project.consphere.dto.CreateCommentRequest;
import com.project.consphere.dto.UpdateCommentRequest;

import java.util.List;

public interface CommentService {

    CommentResponse addComment(Long postId, CreateCommentRequest request);

    List<CommentResponse> getCommentsByPost(Long postId);

    CommentResponse updateComment(Long commentId, UpdateCommentRequest request);

    void deleteComment(Long commentId);
}

File: frontend/src/services/commentService.js
import api from './api';

export const commentService = {
  getCommentsByPost: async (postId) => {
    const response = await api.get(`/comments/post/${postId}`);
    return response.data;
  },

  addComment: async (postId, content) => {
    const response = await api.post(`/comments/post/${postId}`, { content });
    return response.data;
  },

  updateComment: async (commentId, content) => {
    const response = await api.put(`/comments/${commentId}`, { content });
    return response.data;
  },

  deleteComment: async (commentId) => {
    const response = await api.delete(`/comments/${commentId}`);
    return response.data;
  }
};
